src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -379,6 +386,11 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 							
 						break;
 					case view:		
+						if (!getOptionsParamApi().isDisableKey() && !getOptionsParamApi().isNoKeyForSafeOps()) {
```

Comment:
Could `isDisableKey` be moved to `hasValidKey`? (would avoid the need for the callers to check if it's disabled or not.)